x-pack/legacy/plugins/integrations_manager/server/integrations/handlers.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Request, InstallationSavedObject } from '../../common/types';
+import { ArchiveEntry, pathParts } from '../registry';
+import { getClient } from '../saved_objects';
+import { getIntegrations, getIntegrationInfo, installAssets, removeInstallation } from './data';
+
+interface PackageRequest extends Request {
+  params: {
+    pkgkey: string;
+  };
+}
+
+interface InstallAssetRequest extends PackageRequest {
+  params: PackageRequest['params'] & {
+    asset: string;
+  };
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface DeleteAssetRequest extends InstallAssetRequest {}
```

Comment:
Are we expecting these to diverge at some point?

Reply:
Not sure if I follow, but I made some changes here. Can you look at https://github.com/elastic/kibana/pull/40685/commits/13cef20c5c90936c5c494d9e917e37d40b25dfc4 and let me know if you still have questions/suggestions?